### DISCUSSION TITLE
Active Directory as Datastore in openAM

### DISCUSSION THREAD

**Discussion by HarshaMuthuganneSudhakara** [OpenIdentityPlatform/OpenAM#382] [Q&A] [answered]
Question 1) Does changing password of a AD user from openAM change user's AD password?
2) if answer is NO for question 1 then where will be the password stored?
3) Does openAM calls AD or it imports the USER data from AD and does both authentication and authorisation within openAM?

**Comment by maximthomas** [comment, marked as answer]
>  1) Does changing password of a AD user from openAM change user's AD password?

Yes, OpenAM changes AD user password attribute

> 3) Does openAM calls AD or it imports the USER data from AD and does both authentication and authorisation within openAM?

OpenAM calls AD directly, see https://github.com/OpenIdentityPlatform/OpenAM/wiki/How-To-Setup-Active-Directory-Authentication-In-OpenAM

**Reply by HarshaMuthuganneSudhakara** [reply]
Thank you for the response. That helps!!
